internal/backend/telemetry/telemetry.go

```diff
@@ -0,0 +1,143 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/configset"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	api "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"go.uber.org/zap"
+)
+
+type Config struct {
+	Enabled     bool   `koanf:"enabled"`
+	ServiceName string `koanf:"service_name"`
+	Endpoint    string `koanf:"endpoint"`
+}
+
+var Configs = configset.Set[Config]{
+	Default: &Config{Enabled: true, ServiceName: "ak", Endpoint: "localhost:4318"},
+	Dev:     &Config{Enabled: false, ServiceName: "ak", Endpoint: "localhost:4318"},
+}
+
+func toMetricAttrs(attrs map[string]string) []attribute.KeyValue {
+	var result []attribute.KeyValue
+	for k, v := range attrs {
+		result = append(result, attribute.String(k, v))
+	}
+	return result
+}
+
+type Metric interface {
+	Update(value int64, attrs map[string]string)
+}
+
+type Counter struct {
+	counter api.Int64Counter
+}
+
+type UpDownCounter struct {
+	counter api.Int64UpDownCounter
+}
+
+func (m Counter) Update(value int64, attrs map[string]string) {
+	// REVIEW: should we check/report if we get a negative value
+	if enabled {
+		m.counter.Add(context.Background(), value, api.WithAttributes(toMetricAttrs(attrs)...))
+	}
+}
+
+func (m UpDownCounter) Update(value int64, attrs map[string]string) {
+	if enabled {
+		m.counter.Add(context.Background(), value, api.WithAttributes(toMetricAttrs(attrs)...))
+	}
+}
+
+var enabled = false
+
+func metricName(field reflect.Value, fieldType reflect.StructField) string {
+	metricName := fieldType.Name
+	metricName = strings.ReplaceAll(metricName, "_", ".")
+	metricName = "ak." + strings.ToLower(metricName)
+	switch field.Interface().(type) {
+	case Counter:
+		metricName = metricName + ".counter"
```

Comment:
why do we need to add these suffixes?

Reply:
plz see datadog and open telemetry metric naming conventions
https://opentelemetry.io/docs/specs/semconv/general/metrics/
usually it's <service>.<module>.<metric>.<type>

Reply:
i can live with that.